topoflow/components/glacier_energy_balance.py

```diff
@@ -0,0 +1,778 @@
+        'glacier__initial_domain_integral_of_liquid-equivalent_depth': 'm3',
+        'glacier__domain_integral_of_liquid-equivalent_depth': 'm3',
+        'glacier__energy-per-area_cold_content': 'J m-2',
+        'glacier_ice__thickness': 'm',
+        'glacier_ice__initial_thickness': 'm',
+        'glacier__initial_liquid_equivalent_depth': 'm',
+        'glacier_ice__melt_volume_flux': 'm s-1',
+        'glacier_ice__mass-per-volume_density': 'kg m-3',
+        'glacier_ice__mass-specific_isobaric_heat_capacity': 'J kg-1 K-1',
+        'cryosphere__melt_volume_flux': 'm s-1' 
+          }
+
+    #-------------------------------------------------------------------
+    def get_component_name(self):
+  
+        return 'TopoFlow_Glacier_Energy_Balance'
+
+    #   get_component_name()      
+    #-------------------------------------------------------------------
+    def get_attribute(self, att_name):
+
+        try:
+            return self._att_map[ att_name.lower() ]
+        except:
+            print('###################################################')
+            print(' ERROR: Could not find attribute: ' + att_name)
+            print('###################################################')
+            print(' ')
+
+    #   get_attribute()
+    #-------------------------------------------------------------------
+    def get_input_var_names(self):
+
+        #--------------------------------------------------------
+        # Note: These are currently variables needed from other
+        #       components vs. those read from files.
+        #--------------------------------------------------------   
+        return self._input_var_names
+    
+    #   get_input_var_names()
+    #-------------------------------------------------------------------
+    def get_output_var_names(self):
+ 
+        return self._output_var_names
+    
+    #   get_output_var_names()
+    #-------------------------------------------------------------------
+    def get_var_name(self, long_var_name):
+            
+        return self._var_name_map[ long_var_name ]
+
+    #   get_var_name()
+    #-------------------------------------------------------------------
+    def get_var_units(self, long_var_name):
+
+        return self._var_units_map[ long_var_name ]
+   
+    #   get_var_units()
+    #-------------------------------------------------------------------
+    def check_input_types(self):
+
+        #--------------------------------------------------
+        # Notes: rho_H2O, Cp_snow, rho_air, Cp_air,
+        # rho_ice, and Cp_ice are currently always scalars.
+        #--------------------------------------------------        
+        are_scalars = np.array([
+                          self.is_scalar('P_snow'),
+                          self.is_scalar('rho_H2O'),
+                          self.is_scalar('rho_air'),
+                          self.is_scalar('Cp_air'),
+                          self.is_scalar('T_air'),   ##### CHECK THIS ONE.
```

Comment:
what do you mean by "CHECK THIS ONE"??   check unit?

Reply:
This is left over from Scott's snow code, not sure what he meant here

Reply:
Ah, ok.  then no worries (I guessed not very important, but the comment like this just made me wonder)